### ISSUE TITLE
Wallet not reflecting the actual balance

### ISSUE THREAD

**Issue by jeffhappily** [cardano-foundation/cardano-wallet#3136]
### Just checking...

- [X] This is a `cardano-wallet` bug.
- [X] I am using the latest `cardano-wallet` release.
- [X] I am using the *correct* `cardano-node` version for that release of `cardano-wallet`.

### Version

v2022-01-18 (git revision: a5085acbd2670c24251cf8d76a4e83c77a2679ba)

### Platform

Linux jeffhappily 5.13.0-28-generic #31~20.04.1-Ubuntu SMP Wed Jan 19 14:08:10 UTC 2022 x86_64 GNU/Linux

### Installation method

Binary from GitHub release page

### Network configuration

mainnet

## Steps to Reproduce
1. Start the wallet (with the node running already)
```
> cardano-wallet serve --mainnet --node-socket $HOME/cardano/db2/node.socket --database db-wallet +RTS -N4 -RTS
```
2. Query
```
> cardano-wallet wallet get <my id>                                    
{
    "balance": {
        "total": {
            "quantity": 0,
            "unit": "lovelace"
        },
        "available": {
            "quantity": 0,
            "unit": "lovelace"
        },
        "reward": {
            "quantity": 0,
            "unit": "lovelace"
        }
    },
    "name": "...",
    "id": "...",
    "tip": {
        "height": {
            "quantity": 6882893,
            "unit": "block"
        },
        "epoch_number": 321,
        "time": "2022-02-14T22:19:40Z",
        "absolute_slot_number": 53310889,
        "slot_number": 2089
    },
    "passphrase": {
        "last_updated_at": "2022-02-14T16:11:59.411505681Z"
    },
    "address_pool_gap": 20,
    "state": {
        "status": "ready"
    },
    "delegation": {
        "next": [],
        "active": {
            "status": "not_delegating"
        }
    },
    "assets": {
        "total": [],
        "available": []
    }
}
```

### Expected behavior

My actual balance should be shown.

### Actual behavior

I was trying to use `cardano-wallet` to check the balance of my wallet (and perform some transactions), but the correct balance isn't shown. If I query from Cardano Scan or login to Yoroi with the same credentials, I am able to see my correct balance, so I'm not sure what went wrong.

One thing to mention is that when I try to query my addresses using
```
cardano-wallet address list <my id>
```
I can't seem to find my used address (the one that I transferred money to), but when I look up for any of these new addresses on Cardano Scan, it's connected to my previous transactions via the same stake key.

**Comment by HeinrichApfelmus** [comment]
The wallet is fully synced? 🤔 You can get the synchronization status with `GET /v2/network/information` (field `sync_progress`).

**Comment by piotr-iohk** [comment]
> I was trying to use cardano-wallet to check the balance of my wallet (and perform some transactions), but the correct balance isn't shown. If I query from Cardano Scan or login to Yoroi with the same credentials, I am able to see my correct balance, so I'm not sure what went wrong.

Just to make sure, is the wallet that you were using on Yoroi a Shelley wallet? Yoroi used to support also `sequential` Icarus Byron wallet's. Note that if your addresses in the wallet are prefixed with `addr_` then it is a Shelley wallet, otherwise it is most likely Icarus. 

Icarus type of wallet can be restored in cardano-wallet with different endpoint:

[POST /byron-wallets](https://input-output-hk.github.io/cardano-wallet/api/edge/#operation/postByronWallet)

For instance:
```bash
$ curl -X POST http://localhost:8090/v2/byron-wallets \
-d '{"name":"My Test Wallet (icarus)",
      "style":"icarus",
      "passphrase":"Secure Passphrase",
      "mnemonic_sentence":["boil","predict","dolphin","eye","soon","knee","fine","alter","sausage","inquiry","asset","also"]}' \
-H "Content-Type: application/json"
```
Then you can manage transactions on that wallet with this set of endpoints https://input-output-hk.github.io/cardano-wallet/api/edge/#tag/Byron-Transactions.

**Comment by jeffhappily** [comment]
@HeinrichApfelmus Yes it's fully synced
@piotr-iohk It's a Shelley wallet

I'm not sure what happened, I was guessing it _could_ be that the address I transferred to isn't in the list of addresses managed by `cardano-wallet`, so possibly `cardano-wallet` only looked up for those addresses? I don't know, but I managed to resolve this by transferring the fund (via Yoroi) to another `cardano-wallet` managed wallet, and everything works fine. So I'm closing this issue.

Thanks for helping.
